archive/walk.go

```diff
@@ -2,18 +2,232 @@ package archive
 
 import (
 	"context"
+	"fmt"
 	"io"
+	"path"
+	"regexp"
+	"sort"
+	"strconv"
 	"strings"
+	"time"
 
 	"github.com/brimsec/zq/driver"
+	"github.com/brimsec/zq/pkg/byteconv"
 	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/pkg/nano"
 	"github.com/brimsec/zq/scanner"
 	"github.com/brimsec/zq/zbuf"
 	"github.com/brimsec/zq/zio"
 	"github.com/brimsec/zq/zio/detector"
 	"github.com/brimsec/zq/zng/resolver"
+	"github.com/segmentio/ksuid"
 )
 
+const (
+	dataDirname = "zd"
+	zarExt      = ".zar"
+)
+
+// A fileKind is the first part of a file name, used to differentiate files
+// when they are listed from the archive's backing store.
+type fileKind string
+
+const (
+	unknownFileKind fileKind = ""
+	ingestDataKind           = "d"
+	seekIndexKind            = "ts"
```

Comment:
I think these should start with `fileKind`. Will read easier.
```suggestion
	fileKindUnkown fileKind = ""
	fileKindData                    = "d"
	fileKindSeekIndex          = "ts"
```